main.go

```diff
@@ -277,41 +277,45 @@ func fileHasLicense(path string) (bool, error) {
 	return hasLicense(b) || isGenerated(b), nil
 }
 
+// licenseHeader populates the provided license template with data, and returns
+// it with the proper prefix for the file type specified by path. The file does
+// not need to actually exist, only its name is used to determine the prefix.
 func licenseHeader(path string, tmpl *template.Template, data licenseData) ([]byte, error) {
-	var lic []byte
-	var err error
-	switch fileExtension(path) {
-	default:
-		return nil, nil
+	base := strings.ToLower(filepath.Base(path))
+
+	switch fileExtension(base) {
 	case ".c", ".h", ".gv":
-		lic, err = executeTemplate(tmpl, data, "/*", " * ", " */")
+		return executeTemplate(tmpl, data, "/*", " * ", " */")
 	case ".js", ".mjs", ".cjs", ".jsx", ".tsx", ".css", ".scss", ".sass", ".tf", ".ts":
-		lic, err = executeTemplate(tmpl, data, "/**", " * ", " */")
+		return executeTemplate(tmpl, data, "/**", " * ", " */")
 	case ".cc", ".cpp", ".cs", ".go", ".hcl", ".hh", ".hpp", ".java", ".m", ".mm", ".proto", ".rs", ".scala", ".swift", ".dart", ".groovy", ".kt", ".kts", ".v", ".sv":
-		lic, err = executeTemplate(tmpl, data, "", "// ", "")
+		return executeTemplate(tmpl, data, "", "// ", "")
 	case ".py", ".sh", ".yaml", ".yml", ".dockerfile", "dockerfile", ".rb", "gemfile", ".tcl", ".bzl":
-		lic, err = executeTemplate(tmpl, data, "", "# ", "")
+		return executeTemplate(tmpl, data, "", "# ", "")
 	case ".el", ".lisp":
-		lic, err = executeTemplate(tmpl, data, "", ";; ", "")
+		return executeTemplate(tmpl, data, "", ";; ", "")
 	case ".erl":
-		lic, err = executeTemplate(tmpl, data, "", "% ", "")
+		return executeTemplate(tmpl, data, "", "% ", "")
 	case ".hs", ".sql", ".sdl":
-		lic, err = executeTemplate(tmpl, data, "", "-- ", "")
+		return executeTemplate(tmpl, data, "", "-- ", "")
 	case ".html", ".xml", ".vue":
-		lic, err = executeTemplate(tmpl, data, "<!--", " ", "-->")
+		return executeTemplate(tmpl, data, "<!--", " ", "-->")
 	case ".php":
-		lic, err = executeTemplate(tmpl, data, "", "// ", "")
+		return executeTemplate(tmpl, data, "", "// ", "")
 	case ".ml", ".mli", ".mll", ".mly":
-		lic, err = executeTemplate(tmpl, data, "(**", "   ", "*)")
+		return executeTemplate(tmpl, data, "(**", "   ", "*)")
 	}
-	return lic, err
+
+	return nil, nil
 }
 
+// fileExtension returns the file extension of name, or the full name if there
+// is no extension.
 func fileExtension(name string) string {
 	if v := filepath.Ext(name); v != "" {
-		return strings.ToLower(v)
+		return v
```

Comment:
why not lowercase here, once, to avoid forcing all downstream users to do that? Are there one or more downstream users that need extension case preserved?